slash/react/src/ModalAgent/Modal.tsx

```diff
@@ -21,13 +35,25 @@ const Modal = forwardRef<HTMLDialogElement, ModalProps>(
       onOutsideTap,
       children,
       classModifier,
+      size,
       ...props
     }: ModalProps,
     ref,
   ) => {
+    // If size is set to 'lg' or 'sm', use it as the classModifier, otherwise use the provided classModifier
+    let effectiveClassModifier: string | undefined;
+    if (size) {
+      if (classModifier && classModifier !== "lg" && classModifier !== "sm") {
+        effectiveClassModifier = `${size} ${classModifier}`;
+      } else {
+        effectiveClassModifier = size;
+      }
+    } else {
+      effectiveClassModifier = classModifier;
+    }
```

Comment:
Peut-être définir la valeur de la variable effectiveClassModifier tout de suite plutôt que de la calculer à chaque rerender ?

```suggestion
    const effectiveClassModifier = useMemo(() => {
      if (!size) {
       return classModifier;
      }
      if (classModifier && classModifier !== "lg" && classModifier !== "sm") {
        return `${size} ${classModifier}`;
      }
      return size;
    }, [classModifier, size]);
```

Reply:
Le calcul est relativement simple. Ça ne vaut probablement pas le coup de faire un mémo. 
https://maxrozen.com/understanding-when-use-usememo

Reply:
Je dois avouer que dans ce cas précis je serais plus de l'avis de @JLou le calcule et pas complexe ce composant et un composant de fin de chaine donc les rerender vont être localisé sur la modal.
Au pire j'utiliserais une ref pour effectiveClassModifier pour limiter l'impact mémoire mais encore je suis pas trop sur.

par contre ta conditions `classModifier !== "lg" && classModifier !== "sm"` me semble pas bonne car il peut y avoir plusieurs modifier dont des customs du consommateur il ne faudrait pas faire `classModifier.includes` du coup ?

Reply:
c'est trop complexe a gerer. Tant pis pour ceux qui mettent plusieurs modifiers. Ca veut dire extraire les modifiers differents de lg/sm. Je garde tel quel pour l'instant

Reply:
D'accord 👍